### ISSUE TITLE
Objects can be unavailable with requests to different nodes. Tree-service was not distributed to other nodes.

### ISSUE THREAD

**Issue by anatoly-bogatyrev** [nspcc-dev/neofs-node#1732]
On latest master such issue has been detected: 
- put objects by s3 protocol (s3-gw + aws cli) to container with policy: REP 2 CBF 2 SELECT 2.
- objects is available only from two of the four nodes.
- Changing the tree-service endpoint in s3-gw makes the object available

```
admin@jumphost:~/xk6-neofs$ aws --no-verify-ssl s3api head-object --bucket d4bda64e-cf96-487f-96ad-bf21491705b3 --key 4faaca42-a7d0-4306-8087-552d440a368e --endpoint-url http://node4.neofs:8084
An error occurred (404) when calling the HeadObject operation: Not Found


admin@jumphost:~/xk6-neofs$ aws --no-verify-ssl s3api head-object --bucket d4bda64e-cf96-487f-96ad-bf21491705b3 --key 4faaca42-a7d0-4306-8087-552d440a368e --endpoint-url http://node3.neofs:8084
{
    "LastModified": "2022-08-25T16:25:11+00:00",
    "ContentLength": 9216000,
    "ETag": "b366578b4c68c8e9409900d6e8d626b0f4a3bdb86756227908b096eb957c5943",
    "ContentType": "application/octet-stream",
    "Metadata": {}
}
```

```
rpc_endpoint: http://node4.neofs:40332
tree:
  service: node3.neofs:8080
wallet:
```

```
admin@jumphost:~/xk6-neofs$ aws --no-verify-ssl s3api head-object --bucket d4bda64e-cf96-487f-96ad-bf21491705b3 --key 4faaca42-a7d0-4306-8087-552d440a368e --endpoint-url http://node3.neofs:8084
{
    "LastModified": "2022-08-25T16:25:11+00:00",
    "ContentLength": 9216000,
    "ETag": "b366578b4c68c8e9409900d6e8d626b0f4a3bdb86756227908b096eb957c5943",
    "ContentType": "application/octet-stream",
    "Metadata": {}
}
admin@jumphost:~/xk6-neofs$ aws --no-verify-ssl s3api head-object --bucket d4bda64e-cf96-487f-96ad-bf21491705b3 --key 4faaca42-a7d0-4306-8087-552d440a368e --endpoint-url http://node4.neofs:8084
{
    "LastModified": "2022-08-25T16:25:11+00:00",
    "ContentLength": 9216000,
    "ETag": "b366578b4c68c8e9409900d6e8d626b0f4a3bdb86756227908b096eb957c5943",
    "ContentType": "application/octet-stream",
    "Metadata": {}
}
```

**Comment by anatoly-bogatyrev** [comment]
The issue occurs only(!) when using the preset script ([link](https://github.com/nspcc-dev/xk6-neofs/blob/master/scenarios/preset/preset_s3.py)) in write via aws cli in more than 1 python thread.

For example in case of uploading via K6 issue has not been reproduced:
```
INFO[0011] > 9dde0081-aeff-49e4-b230-d23b0afe53fd - a2b0396a-f438-4069-aa0a-a76a7641355a  source=console
INFO[0011] > 861ecddb-86ec-4fd8-a024-2223da167fee - e4f7801d-a736-4f77-83ec-e6b9b2f5ad92  source=console
INFO[0011] > e023ae85-c99d-43aa-afb2-89114b376aaf - 6075b676-1326-49d6-b7b8-7c37200ce58c  source=console
INFO[0011] > fc859b66-a220-4a49-8183-1016af0ad1db - d5ac5005-29cf-41d9-868c-eb6f88d80da1  source=console
INFO[0011] > b2251d1d-9edb-43b7-9011-e3d4776002cf - bafa00d1-5c4c-4946-9ed8-a1556951da54  source=console
INFO[0011] > b2251d1d-9edb-43b7-9011-e3d4776002cf - 6b9265f4-321c-43ad-84dd-dd8362c43494  source=console

running (11.7s), 000/400 VUs, 1631 complete and 0 interrupted iterations
write ✓ [======================================] 400 VUs  10s

     █ setup

     aws_obj_put_duration...: avg=2.65s min=499.41ms med=2.24s max=9.16s p(90)=4.76s p(95)=6.16s
     aws_obj_put_total......: 1631  138.96804/s
     data_received..........: 0 B   0 B/s
     data_sent..............: 13 GB 1.1 GB/s
     iteration_duration.....: avg=2.66s min=97.56µs  med=2.25s max=9.16s p(90)=4.76s p(95)=6.16s
     iterations.............: 1631  138.96804/s
     vus....................: 103   min=103     max=400
     vus_max................: 400   min=400     max=400

prod:neofsadmin@jumphost:~/xk6-neofs$ aws s3api get-object --bucket b2251d1d-9edb-43b7-9011-e3d4776002cf --key 6b9265f4-321c-43ad-84dd-dd8362c43494 data-out-copy.txt --endpoint-url http://node1.neofs:8084
{
    "LastModified": "2022-08-31T10:59:07+00:00",
    "ContentLength": 8192000,
    "ETag": "d74685619b8b6bf901ab1757e196b9c510c89a5115a3a62b334c679c1280b204",
    "VersionId": "GTh5LRjn6C6g9kXkVkSy8cgDKRVX1hwDrKQiPHXWtGwo",
    "ContentType": "application/octet-stream",
    "Metadata": {},
    "TagCount": 0
}
prod:neofsadmin@jumphost:~/xk6-neofs$ aws s3api get-object --bucket b2251d1d-9edb-43b7-9011-e3d4776002cf --key 6b9265f4-321c-43ad-84dd-dd8362c43494 data-out-copy.txt --endpoint-url http://node2.neofs:8084
{
    "LastModified": "2022-08-31T10:59:07+00:00",
    "ContentLength": 8192000,
    "ETag": "d74685619b8b6bf901ab1757e196b9c510c89a5115a3a62b334c679c1280b204",
    "VersionId": "GTh5LRjn6C6g9kXkVkSy8cgDKRVX1hwDrKQiPHXWtGwo",
    "ContentType": "application/octet-stream",
    "Metadata": {},
    "TagCount": 0
}
prod:neofsadmin@jumphost:~/xk6-neofs$ aws s3api get-object --bucket b2251d1d-9edb-43b7-9011-e3d4776002cf --key 6b9265f4-321c-43ad-84dd-dd8362c43494 data-out-copy.txt --endpoint-url http://node3.neofs:8084
{
    "LastModified": "2022-08-31T10:59:08+00:00",
    "ContentLength": 8192000,
    "ETag": "d74685619b8b6bf901ab1757e196b9c510c89a5115a3a62b334c679c1280b204",
    "VersionId": "GTh5LRjn6C6g9kXkVkSy8cgDKRVX1hwDrKQiPHXWtGwo",
    "ContentType": "application/octet-stream",
    "Metadata": {},
    "TagCount": 0
}
prod:neofsadmin@jumphost:~/xk6-neofs$ aws s3api get-object --bucket b2251d1d-9edb-43b7-9011-e3d4776002cf --key 6b9265f4-321c-43ad-84dd-dd8362c43494 data-out-copy.txt --endpoint-url http://node4.neofs:8084
{
    "LastModified": "2022-08-31T10:59:07+00:00",
    "ContentLength": 8192000,
    "ETag": "d74685619b8b6bf901ab1757e196b9c510c89a5115a3a62b334c679c1280b204",
    "VersionId": "GTh5LRjn6C6g9kXkVkSy8cgDKRVX1hwDrKQiPHXWtGwo",
    "ContentType": "application/octet-stream",
    "Metadata": {},
    "TagCount": 0
}

```

**Comment by fyrchik** [comment]
Closed via #2018
